lstchain/calib/camera/calibrator.py

```diff
@@ -207,82 +184,39 @@ def _calibrate_dl1(self, event, telid):
         if self._check_dl0_empty(waveforms):
             return
 
-        if self.image_extractor.requires_neighbors():
-            camera = event.inst.subarray.tel[telid].camera
-            self.image_extractor.neighbors = camera.neighbor_matrix_where
+        # for the moment we do the gain selection afterwards
+        # use gain mask without gain selection
 
-        charge, pulse_time = self.image_extractor(waveforms)
+        # TBD: - perform calibration of the R1 waveform (not DL1)
+        #      - gain selection before charge integration
 
-        # correct charge for width integration
-        corrected_charge = charge * self.charge_correction[:,np.newaxis]
+
+        no_gain_selection= np.zeros((waveforms.shape[0],waveforms.shape[1]), dtype=np.int)
+
+        charge, pulse_time = self.image_extractor(waveforms, telid, no_gain_selection)
+
+        # correct charge for global scale
+        corrected_charge = charge * np.array(self.charge_scale)[:,np.newaxis]
 
         # correct time with drs4 correction if available
         if self.time_corrector:
             pulse_time = self.time_corrector.get_corr_pulse(event, pulse_time)
 
         # add flat-fielding time correction
-        pulse_time_ff_corrected = pulse_time + self.mon_data.tel[telid].calibration.time_correction
+        pulse_time_ff_corrected = pulse_time + self.mon_data.tel[telid].calibration.time_correction.value
```

Comment:
how can we insure `pulse_time` and the `time_correction` are in the same unit?

Reply:
The are both calculated by the charge extractor using the waveform. But, you are right, as it is done now we can not be sure. The problem is that the  time do not have units in the R1Container and DL1 container